### ISSUE TITLE
Github is sunsetting Atom

### ISSUE THREAD

**Issue by Zalastax** [tidalcycles/Tidal#920]
Github is [Sunsetting Atom](https://github.blog/2022-06-08-sunsetting-atom/). We need a replacement editor to use in the Tidal installation packages and in our recommendations to new users.

**Comment by ndr-brt** [comment]
The most natural decision would be to move our efforts to improve the vscode extension, but the more I think about this, the more I believe we should go with a dedicated editor.
For example I got little lost with the work on https://github.com/polymorphicengine/tidal-gui, but could be definitely something that we could push to become the default tidal editor.
we should take this atom thing as an opportunity to leverage our toolset

**Comment by yaxu** [comment]
Yes tidal-gui uses codemirror, as does strudel: https://strudel.tidalcycles.org/

The latter supports event highlighting. 

It could be possible to unify this effort.. But maybe that would be a lot of work.

**Comment by ndr-brt** [comment]
> Yes tidal-gui uses codemirror, as does strudel: https://strudel.tidalcycles.org/
> 
> The latter supports event highlighting.
> 
> It could be possible to unify this effort.. But maybe that would be a lot of work.

Our need to pass over this atom thing is to have a replacement for it, no need to have event highlighting for then, just the basic features and an easy installation process, then we could improve that after then.

I would be available to set up a chat/discord channel with the interested ones where we can talk about what we have and what we need by the end of december, and let's see if that's feasible or not. wdyt?

**Comment by yaxu** [comment]
Yes agreed fully! Lets remove barriers to getting something working as soon as we can.

**Comment by polymorphicengine** [comment]
I do think tidal-gui has potential. At this point i have fixed some of the major issues with it (like hard crashes etc.). It is actually the only thing I am using to make music since more than half a year, so I can say that it is quite stable now. I also think that it has pontential to provide features beyond what atom had to offer.
What is also neat, is the possiblity to provide binarys that are not dependend on ghc

**Comment by telephon** [comment]
OT: Isn't "sunsetting" a beautiful euphemism (I can imagine lots of future uses for this word I'd rather not say ...)

**Comment by thgrund** [comment]
I am also very interested in the exchange. Then let's give it a try I would say. Sounds very exciting in any case :-)

**Comment by cleary** [comment]
I think this can be closed now

A summary of discussions around the place:

Short term: pulsar (atom fork) has come to the fore as the best choice as a direct replacement for atom, @ndr-brt has ported his atom-tidalcycles plugin to work, there is more to do with package management support/functionality still but it can be installed manually. @HighHarmonics2 has added some [detailed documentation](https://github.com/tidalcycles/tidal-doc/pull/132)

Medium/long term: @mindofmatthew has made good progress with the[ text.management editor](https://github.com/mindofmatthew/text.management) as a dedicated livecode editor, supporting tidalcycles to begin. There is [explicit interest in multi-language support](https://github.com/mindofmatthew/text.management/issues/34) meeting @fracnesco 's requirement

**Comment by ndr-brt** [comment]
Close as now there's Pulsar and the tidal plugin is working nicely (it changed name already: https://github.com/tidalcycles/pulsar-tidalcycles

**Comment by yaxu** [comment]
The windows installer is still trying to install atom. Thanks @highharmonics2 for looking into this as well as the Mac install!

**Comment by sss-create** [comment]
Shall we make this a discussion or close? I am not up to date in terms of editor choice/installation.

**Comment by yaxu** [comment]
I think we closed it once and can close it again by now..

**Comment by ndr-brt** [comment]
> Shall we make this a discussion or close? I am not up to date in terms of editor choice/installation.

Just for the record, a community driven fork of Atom is alive, it's called Pulsar (https://pulsar-edit.dev/)
The tidalcycles plugin for atom has been updated and adapted to be fully working with it: https://github.com/tidalcycles/pulsar-tidalcycles
